src/Libraries/RevitNodes/GeometryConversion/ProtoToRevitMesh.cs

```diff
@@ -102,35 +101,42 @@ public static class ProtoToRevitMesh
              ElementId MaterialId = null,
             bool performHostUnitConversion = true)
         {
-         
+
             var verts = mesh.VertexPositions;
-            var indicies = mesh.FaceIndices;
+            var indices = mesh.FaceIndices.ToList();
 
             var currentVerts = new List<Autodesk.DesignScript.Geometry.Point>();
             var tsb = new TessellatedShapeBuilder();
             tsb.OpenConnectedFaceSet(false);
 
-            foreach (var f in indicies)
+            for (int faceindex = 0; faceindex < indices.Count(); faceindex++)
             {
-
+                var f = indices[faceindex];
                 currentVerts.Clear();
-                var currentIndicies = new List<uint>() { f.A, f.B, f.C, f.D };
-                for (int i = 0; i < f.Count; i++)
+                var currentIndices = new List<uint>() { f.A, f.B, f.C, f.D };
+
+                //if this is a quad face triangulate it and skip the rest of this loop
+                if (f.Count > 3)
                 {
-                    var currentindex = Convert.ToInt32(currentIndicies[i]);
+                    indices.Add(IndexGroup.ByIndices(f.B, f.C, f.A));
+                    indices.Add(IndexGroup.ByIndices(f.A, f.C, f.D));
+                    continue;
```

Comment:
You should also keep in mind that the `IndexGroup.ByIndices` call actually has a lot of function call overhead.